config/get_git_version.py

```diff
@@ -0,0 +1,93 @@
+# used by the library to report its version.
+import argparse
+import sys
+import os
+import re
+
+
+def normalize_version(git_version, git_is_dirty):
+    if '-' in git_version:
+        cleaned = re.search('[0-9]+\.[0-9]+\.[0-9]\-[0-9]+', git_version)
+        cleaned_string = cleaned.group(0).replace("-", ".")
+    elif 'v' in git_version:
+        cleaned_string = git_version.replace("v", "")
+    else:
+        cleaned_string = git_version
+
+    # In case the repository is in a dirty state (uncommited changes)
+    # we do tell the user during build by writing to stdout.
+    # That is the way it is done in the CMake Build as well.
+    # Maybe think about adding the -dirty also for the version header.
+    if git_is_dirty == "TRUE":
+        git_version_dirty = git_version+"-dirty"
+        print("git version: " + git_version_dirty +
+              " normalized to " + cleaned_string)
+
+    return cleaned_string
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Generate version header')
+    parser.add_argument('--header',
+                        required=True,
+                        help='output header file')
+    parser.add_argument('--volatile_file',
+                        required=True,
+                        help='file containing the git version variables')
+    parser.add_argument('--version_variable_name',
+                        help='variablename of the hash')
+    parser.add_argument('--is_dirty_name',
+                        help='variablename of the boolean communicating if the workspace has no local changes')
+    parser.add_argument('--default_version',
+                        help='variablename for version which should be used in case git was not executable.')
+
+    args = parser.parse_args()
+
+    # Read volatile-status.txt file
+    git_version = ""
+    is_dirty = ""
+    try:
+        with open(args.volatile_file, "r") as f:
+            for entry in f.read().split("\n"):
+                if entry:
+                    key_value = entry.split(' ', 1)
+                    key = key_value[0].strip()
+                    if key == args.version_variable_name:
+                        git_version = key_value[1].strip()
+                    if key == args.is_dirty_name:
+                        is_dirty = key_value[1].strip()
+    except:
+        # In case volatile-status cannot be read, exit with an error
+        sys.exit("Cannot open volatile-status.txt")
+
+    if git_version == "" or is_dirty == "":
+        sys.exit("No usable entry in volatile-status.txt")
+
+    git_version = normalize_version(git_version, is_dirty)
+
+    # In case we werent able to determine the current version
+    # use the default set version
+    if git_version == "0.0.0":
+        git_version = args.default_version
+
+    # Notify the user about the version used.
+    print("Version: " + git_version)
+
+    # Write the actual version.h
+    with open(args.header, "w") as f:
+        f.write("""
```

Comment:
can we use include/version.h.in for this too? i know it's a bit more complex but it means we don't diverge.

Reply:
I guess we can. However this will add a dependency between the bazel build and the CMake build system as then we have a file format which both versions need to be able to process. But I guess for consistency overall its better.
Will do this later

Reply:
Sorry looking at it I don't think we cant really do this. This would need us to figure out how to get to the base repository from the bazel build folder leaving us to the same problems as the implementation with genrule.
We can add /include/version.h.in as a src to gen_version_header.bzl but it is not read or writable then according to 
Files are represented by [File](https://docs.bazel.build/versions/main/skylark/lib/File.html) objects. Since Bazel does not perform file I/O during the analysis phase, these objects cannot be used to directly read or write file content. Rather, they are passed to action-emitting functions (see [ctx.actions](https://docs.bazel.build/versions/main/skylark/lib/actions.html)) to construct pieces of the action graph.
https://docs.bazel.build/versions/main/skylark/rules.html#contents
So I don't see a way of getting this file into get_git_version.py

Reply:
i thought we'd just file open "../include/version.h.in" :D

Reply:
I don't think that works that easy. I've done it the suggested way by passing it as input argument to gen_version_header.bzl